### ISSUE TITLE
Interactive Tutorial - Exploring Your App

### ISSUE THREAD

**Issue by ymartineau** [kubernetes/website#26199]
Module 3 - Step 2

curl http://localhost:8001/api/v1/namespaces/default/pods/$POD_NAME/proxy/

This doesn't work when executed on a local minikube installation on windows (it works as expected on katacoda but the difference between those two environments is not obvious for this specific issue).

I'm using a virtualbox driver (I don't know if it matters).
Should this kind of bug be open against minikube or here?

curl http://localhost:8001/api/v1/namespaces/default/pods/kubernetes-bootcamp-57978f5f5d-pjl9r/proxy/
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {

  },
  "status": "Failure",
  "message": "error trying to reach service: dial tcp 172.17.0.3:80: connect: connection refused",
  "code": 500
}

Here the connection attempt is made on port 80 while NodeJs is running on port 8080.

**Comment by sftim** [comment]
This looks like a report for https://kubernetes.io/docs/tutorials/kubernetes-basics/explore/explore-interactive/

Make sure that when you run `kubectl proxy` it stays running (you don't quit the proxy and then try to use it to access pods). The advice is to open a second window (PowerShell, etc) to run the remaining commands.

If you see an error from `kubectl proxy` then please describe it.

Can you confirm the details of the system that someone could replicate this on @ymartineau ?
- Windows version
- minikube version
- Kubernetes version
- VirtualBox version
- Processor architecture, I'm assuming AMD64

plus anything else that seems relevant.

**Comment by ymartineau** [comment]
Thank you for your response.

Yes, this is the right url.

Yes I kept `kubectl proxy` running in one window (cmd prompt, not powershell, but I'm not sure it matters) (there was no output message) and ran curl in another window (cmd prompt).

windows version: windows 10 x64 patch 1809
minikube version: v1.16.0

Kubernetes:
kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:28:09Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.0", GitCommit:"af46c47ce925f4c4ad5cc8d1fca46c7b77d13b38", GitTreeState:"clean", BuildDate:"2020-12-08T17:51:19Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}

Virtualbox Version 6.1.16 r140961

Processor architecture: x64

thank you

**Comment by ch4mpy** [comment]
Same `kubectl proxy` issue here with with Minikube 1.17.1 and VirtualBox 6.1.18 r142142 running on Windows 10.

Running `kubectl exec -ti [my-running-pod] bash` and then `curl http://localhost:8080/api/v1/namespaces/default/pods/kubernetes-bootcamp-57978f5f5d-nddqv/proxy/` works.

Running `kubectl proxy` gives `Starting to serve on 127.0.0.1:8001` but then accessing http://localhost:8001/api/v1/namespaces/default/pods/kubernetes-bootcamp-57978f5f5d-nddqv/proxy/ returns:
```
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {
    
  },
  "status": "Failure",
  "message": "error trying to reach service: dial tcp 172.17.0.3:80: connect: connection refused",
  "code": 500
}
```

**Comment by Karmavil** [comment]
@ymartineau You're right. 

The exercise doesn't assign a port in the deployment. 
Try:
`kubectl create deployment kubernetes-bootcamp --image=gcr.io/google-samples/kubernetes-bootcamp:v1 --port=8080`
Using a new tab or terminal run `kubectl proxy --port=8080`
Define your $POD_NAME or just copy the name from the `kubectl get pods` output and run the curl command. It will work

**Comment by simonc6372** [comment]
I ran into this too, and was about to create an issue, but found this one. 

The deployment created in the preceding lesson (https://kubernetes.io/docs/tutorials/kubernetes-basics/deploy-app/deploy-interactive/) does not have a port assigned. 
This tutorial assumes the deployment *does* have a port assigned, but if you are following the lessons against your own cluster, this is not the case. 
This makes it hard to follow along against your own cluster. 
I think you either need to tweak the image, or the tutorial command to assign the port.

**Comment by sftim** [comment]
@ymartineau would you be willing to summarize https://github.com/kubernetes/website/issues/26199#issuecomment-787444558 into the description of this issue? I think that'd help.

**Comment by ymartineau** [comment]
hi, FYI I tried the instructions provided by @Karmavil, the curl command to get the api version (curl http://localhost:8001/version) had to be updated to use the 8080 port (curl http://localhost:8080/version) and it worked. Thank you

I was intrigued by the 8080 port so I tried the following:

kubectl create deployment kubernetes-bootcamp --image=gcr.io/google-samples/kubernetes-bootcamp:v1 --port=8001
kubectl proxy

and it didn't work with same error message as original attempt.
So I "guess" there is something special with 8001 port. This is not necessarily the point here, so as mentioned I guess something could be updated in the tutorial or the image to be consistent.

Thank you

**Comment by Karmavil** [comment]
/team/katacoda

**Comment by Karmavil** [comment]
@sftim I do not know how to proceed with this issue and this is why:

* The minikube cluster (*interactive tutorial*) is actually opening a **containerPort**, you can see it is available in the step 1 of the module 3 `kubectl describe pods` (after the `kubectl create ...` command is executed in the previous module 2). But if you try to create the same deployment in your local cluster `kubectl create ...`  it does not assign a **containerPort**

The simplest way to fix this would be with a note: 
> If you're running this tutorial in your local cluster run this command instead ... 

But something is happening in the process that I still cannot yet understand, sorry I thought it was related with an outdated version of minikube but apparently it is not. I cannot give better assistance on this issue.

Maybe the problem has something to do with this :
```bash
# outputs from interactive tutorial
$ minikube version
minikube version: v1.8.1

$ kubectl get nodes
NAME       STATUS   ROLES    AGE   VERSION
minikube   Ready    master   17m   v1.17.3
```

**Comment by ymartineau** [comment]
adding a note would be an acceptable first step:

> If you're running this tutorial in your local cluster run this command instead ...

**Comment by Karmavil** [comment]
Thanks for the feedback @ymartineau I will try it (soon). 
I'm not sure if the exercise is embedded so I can't confirm if that's something easy to do. Also there is a release incoming so people is busy and I don't want to brake thing right now. 
I will take a look

**Comment by Karmavil** [comment]
All right guys I sent a message to address this issue.
There is [an issue](https://github.com/katacoda-scenarios/kubernetes-bootcamp-scenarios/issues/18) currently open so I will not create a new one.
There is no need to create a pull request because the member of that project already made this changes in another repo so I asked him to merge that changes
> The problem is triggered in module 2 when you create the deploy.. but he will understand

**Comment by sftim** [comment]
https://github.com/kubernetes/website/issues/26199#issuecomment-801716134 seems like a reasonable fix to me!

**Comment by AdamHollow49** [comment]
Ran into this today - comments here saved it! No note added to tutorial yet?

**Comment by sftim** [comment]
@AdamHollow49 if you'd like to [propose](https://kubernetes.io/docs/contribute/) a fix, the Kubernetes special interest group for documentation can help you do that.

Like Kubernetes itself, the documentation is open source. Most improvements are done by volunteers.
/remove-lifecycle stale
